src/SFA.DAS.FindEpao.UnitTests/Application/Courses/Queries/GetCourseEpao/WhenHandlingGetCourseEpaoQuery.cs

```diff
@@ -91,6 +91,58 @@ public class WhenHandlingGetCourseEpaoQuery
         }
 
+        [Test, MoqAutoData]
+        public async Task If_No_Valid_Standards_Found_Returns_Zero_AllCourses(
+            GetCourseEpaoQuery query,
+            GetEpaoResponse epaoApiResponse,
+            List<GetCourseEpaoListItem> courseEpaosApiResponse,
+            List<GetEpaoCourseListItem> epaoCoursesApiResponse,
+            List<GetDeliveryAreaListItem> areasFromCache,
+            GetStandardsListResponse coursesFromCache,
+            [Frozen] Mock<IAssessorsApiClient<AssessorsApiConfiguration>> mockAssessorsApiClient,
+            [Frozen] Mock<ICachedDeliveryAreasService> mockCachedAreasService,
+            [Frozen] Mock<ICachedCoursesService> mockCachedCoursesService,
+            [Frozen] Mock<ICourseEpaoIsValidFilterService> mockCourseEpaoFilter,
+            GetCourseEpaoQueryHandler handler)
+        {
+            courseEpaosApiResponse[0].EpaoId = query.EpaoId.ToLower();
+            coursesFromCache.Standards.First().LarsCode = epaoCoursesApiResponse.First().StandardCode;
+            coursesFromCache.Standards.ElementAt(1).LarsCode = query.CourseId;
+            mockAssessorsApiClient
+                .Setup(client => client.Get<GetEpaoResponse>(
+                    It.Is<GetEpaoRequest>(request => request.EpaoId == query.EpaoId)))
+                .ReturnsAsync(epaoApiResponse);
+            mockAssessorsApiClient
+                .Setup(client => client.GetAll<GetCourseEpaoListItem>(
+                    It.Is<GetCourseEpaosRequest>(request => request.CourseId == query.CourseId)))
+                .ReturnsAsync(courseEpaosApiResponse);
+            mockAssessorsApiClient
+                .Setup(client => client.GetAll<GetEpaoCourseListItem>(
+                    It.Is<GetEpaoCoursesRequest>(request => request.EpaoId == query.EpaoId)))
+                .ReturnsAsync(epaoCoursesApiResponse);
+            mockCachedAreasService
+                .Setup(service => service.GetDeliveryAreas())
+                .ReturnsAsync(areasFromCache);
+            mockCachedCoursesService
+                .Setup(client => client.GetCourses())
+                .ReturnsAsync(coursesFromCache);
+            mockCourseEpaoFilter
+                .Setup(service => service.IsValidCourseEpao(It.IsAny<GetCourseEpaoListItem>()))
+                .Returns<GetCourseEpaoListItem>(item => item.EpaoId == query.EpaoId.ToLower());
+            mockCourseEpaoFilter
+                .Setup(service => service.ValidateEpaoStandardDates(It.IsAny<DateTime?>(), It.IsAny<DateTime?>(), It.IsAny<DateTime?>()))
+                .Returns(true);
+            mockCourseEpaoFilter
+                .Setup(service => service.ValidateVersionDates(It.IsAny<DateTime?>(), It.IsAny<DateTime?>()))
+                .Returns(false);
+
+            var result = await handler.Handle(query, CancellationToken.None);
+
+            result.Epao.Should().BeEquivalentTo(epaoApiResponse);
+            result.AllCourses.Should().BeEmpty();
+        }
```

Comment:
Can we add the assertion `result.StandardVersions` to this test and others as it's a new property returned by the result object?

Reply:
Assertion added for StandardVersions